### DISCUSSION TITLE
Alternative to using ResteasyProviderFactory?

### DISCUSSION THREAD

**Discussion by xgp** [keycloak/keycloak#17423] [Q&A]
When creating a custom REST resource, it used to be the standard to use `ResteasyProviderFactory` to inject things like the session or client context. I noticed this was changed in `v21`, but I can't figure out what, if anything, is the recommended way to replace this. Any ideas? Thanks!

```java
  ResteasyProviderFactory.getInstance().injectProperties(resource);
```

**Comment by koplas** [comment]
Were you able to get the `ResteasyProviderFactory` instance?

**Reply by xgp** [reply]
I stopped trying to use it to inject properties, and just updated all my constructors to require the objects. Ultimately, I suppose this is the better solution, but the `ResteasyProviderFactory.getInstance().injectProperties(resource);` was convenient, and it was frustrating not to see that in the migration/upgrade notes.
